comp/core/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,163 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"context"
+	"testing"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCounterInitializer(t *testing.T) {
+
+	telemetry := newMock().(*telemetryImpl)
```

Comment:
Did you consider having a `TelemetryMock` interface so you don't have to rely on the implementation knowledge of `newMock`?